src/peer/identity.js

```diff
@@ -28,25 +27,24 @@ function loadIdentity (filePath: string): Promise<PeerId> {
 }
 
 
-function loadOrGenerateIdentity(filePath: string): PeerId {
-  let peerId
-  try {
-    loadIdentity(filePath)
-  } catch(err) {
-    console.log(`Could not load from ${filePath}, generating new PeerId...`)
-    peerId = generateIdentity()
-    saveIdentity(peerId, filePath)
-  }
-
-  return peerId
+function loadOrGenerateIdentity(filePath: string): Promise<PeerId> {
+    return loadIdentity(filePath)
+      .catch(err => {
+        console.log(`Could not load from ${filePath}, generating new PeerId...`)
+        return generateIdentity()
+      })
+      .then(id => {
+        saveIdentity(id, filePath)
+        return id
+      })
 }
 
-function inflateMultiaddr(multiaddrString: string): PeerInfo {
+function inflateMultiaddr(multiaddrString: string): Promise<PeerInfo> {
     const multiaddr = Multiaddr(multiaddrString)
-    const peerInfo = new PeerInfo()
-    peerInfo.multiaddr.add(multiaddr)
-
-    return peerInfo
+    return PeerInfo.create().then(peerInfo => {
```

Comment:
Ah yeah, I see.  That does seem like a better way to go, and `PeerId.createFromB58String` has the added bonus of being synchronous 😄